### PR TITLE
Added unit tests for BCVRef equality operator.

### PR DESCRIPTION
**Pull request by tombogle**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1099)

### CODE REVIEW

File: SIL.Scripture.Tests/BCVRefTests.cs [hunk 1]
```diff
@@ -553,6 +553,117 @@ public string MakeReferenceString_DifferentBooks(string cvSeparator, string brid
...
+			Assert.IsFalse(new BCVRef(5, 2, 1) == null);
+		}
+
+		/// ------------------------------------------------------------------------------------
+		/// <summary>
+		/// Tests the EqualityOperator when the first reference is invalid
+		/// </summary>
+		/// ------------------------------------------------------------------------------------
+		[TestCase(-1, 2, 3)]
+		[TestCase(1, 3, 2)]
```
Comment (ermshiperete) [inline SIL.Scripture.Tests/BCVRefTests.cs:597]: Why is this an invalid reference? (similar in the other tests below)

**Review by ermshiperete — COMMENTED** [review]
:lgtm:, but I find the naming of the tests a bit confusing (or one of the test cases on most test methods)

File: SIL.Scripture.Tests/BCVRefTests.cs (hunk 1 above)
Comment (tombogle) [reply SIL.Scripture.Tests/BCVRefTests.cs]: It wasn't. That was me working too fast on a sleep-deprived brain. Fixed it now.

**Comment by tombogle** [comment]
> :lgtm:, but I find the naming of the tests a bit confusing (or one of the test cases on most test methods)

I'm not sure if this comment refers to the bad test cases (which I have now fixed) or something else.

**Review by FoolRunning — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1099#-:-MkgM_ss-fb8NOvRzgUw:bg7igkv)* status: all files reviewed, 1 unresolved discussion (waiting on @ermshiperete)

File: SIL.Scripture.Tests/BCVRefTests.cs [hunk 2]
```diff
@@ -553,6 +553,119 @@ public string MakeReferenceString_DifferentBooks(string cvSeparator, string brid
...
+			Assert.IsFalse(new BCVRef(5, 2, 1) == new BCVRef(book, chapter, verse));
+		}
+
+		/// ------------------------------------------------------------------------------------
+		/// <summary>
+		/// Tests the EqualityOperator when the both references are invalid but equivalent
+		/// </summary>
+		/// ------------------------------------------------------------------------------------
+		[TestCase(-1, 2, 3)]
+		[TestCase(1, 3, 2)]
```
Comment (ermshiperete) [inline SIL.Scripture.Tests/BCVRefTests.cs:625]: I think here's another test case that doesn't match what the test name suggests

**Comment by ermshiperete** [comment]
> > :lgtm:, but I find the naming of the tests a bit confusing (or one of the test cases on most test methods)
> 
> I'm not sure if this comment refers to the bad test cases (which I have now fixed) or something else.

yes, it referred to the bad tests cases - I couldn't tell if the test case was bad, or the test method name.

**Review by ermshiperete — CHANGES_REQUESTED** [review]

**Review by ermshiperete — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1099)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @tombogle)

**Pull request merged**